{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/workspace/notebooks', '/opt/conda/lib/python37.zip', '/opt/conda/lib/python3.7', '/opt/conda/lib/python3.7/lib-dynload', '', '/opt/conda/lib/python3.7/site-packages', '/opt/conda/lib/python3.7/site-packages/IPython/extensions', '/home/jovyan/.ipython', '../ngsutils/', '../ngsutils/', '../ngsutils/', '../ngsutils/', '../ngsutils/reader']\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "sys.path.append(\"../ngsutils/\")\n",
    "sys.path.append(\"../ngsutils/reader\")\n",
    "print(sys.path)\n",
    "\n",
    "import anndata\n",
    "import pandas as pd\n",
    "import reader"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "path_to_featureCounts_data = \"../data/read_featureCounts/featureCounts_sample.txt\"\n",
    "path_to_conditions = \"../data/read_featureCounts/conditions.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = reader.read_featureCounts(path_to_featureCounts_data, path_to_conditions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Geneid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MapolyY_A0001</th>\n",
       "      <td>Chr_Y_A</td>\n",
       "      <td>10168</td>\n",
       "      <td>10935</td>\n",
       "      <td>-</td>\n",
       "      <td>768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_A0002</th>\n",
       "      <td>Chr_Y_A</td>\n",
       "      <td>14752</td>\n",
       "      <td>16054</td>\n",
       "      <td>-</td>\n",
       "      <td>1303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_A0003</th>\n",
       "      <td>Chr_Y_A</td>\n",
       "      <td>69760</td>\n",
       "      <td>71066</td>\n",
       "      <td>-</td>\n",
       "      <td>1307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_A0004</th>\n",
       "      <td>Chr_Y_A;Chr_Y_A;Chr_Y_A;Chr_Y_A;Chr_Y_A;Chr_Y_A</td>\n",
       "      <td>74668;75039;75534;75986;84872;84872</td>\n",
       "      <td>74870;75338;75600;77256;86506;86506</td>\n",
       "      <td>+;+;+;+;+;+</td>\n",
       "      <td>3476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_A0005</th>\n",
       "      <td>Chr_Y_A</td>\n",
       "      <td>125980</td>\n",
       "      <td>126717</td>\n",
       "      <td>-</td>\n",
       "      <td>738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0099s0063</th>\n",
       "      <td>scaffold_99;scaffold_99;scaffold_99;scaffold_99</td>\n",
       "      <td>770737;771728;772563;773007</td>\n",
       "      <td>771554;772149;772759;775346</td>\n",
       "      <td>-;-;-;-</td>\n",
       "      <td>3777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0991s0001</th>\n",
       "      <td>scaffold_991</td>\n",
       "      <td>5953</td>\n",
       "      <td>6171</td>\n",
       "      <td>-</td>\n",
       "      <td>219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0992s0001</th>\n",
       "      <td>scaffold_992;scaffold_992;scaffold_992;scaffol...</td>\n",
       "      <td>11;547;996;1312;1839</td>\n",
       "      <td>255;712;1187;1652;1883</td>\n",
       "      <td>-;-;-;-;-</td>\n",
       "      <td>989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0994s0001</th>\n",
       "      <td>scaffold_994</td>\n",
       "      <td>4875</td>\n",
       "      <td>6206</td>\n",
       "      <td>-</td>\n",
       "      <td>1332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0998s0001</th>\n",
       "      <td>scaffold_998;scaffold_998;scaffold_998;scaffol...</td>\n",
       "      <td>1631;1804;2065;2469;2959;3516;3881;4406;4625</td>\n",
       "      <td>1690;1859;2102;2805;3334;3733;4085;4588;4861</td>\n",
       "      <td>+;+;+;+;+;+;+;+;+</td>\n",
       "      <td>1710</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19287 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               Chr  \\\n",
       "Geneid                                                               \n",
       "MapolyY_A0001                                              Chr_Y_A   \n",
       "MapolyY_A0002                                              Chr_Y_A   \n",
       "MapolyY_A0003                                              Chr_Y_A   \n",
       "MapolyY_A0004      Chr_Y_A;Chr_Y_A;Chr_Y_A;Chr_Y_A;Chr_Y_A;Chr_Y_A   \n",
       "MapolyY_A0005                                              Chr_Y_A   \n",
       "...                                                            ...   \n",
       "Mapoly0099s0063    scaffold_99;scaffold_99;scaffold_99;scaffold_99   \n",
       "Mapoly0991s0001                                       scaffold_991   \n",
       "Mapoly0992s0001  scaffold_992;scaffold_992;scaffold_992;scaffol...   \n",
       "Mapoly0994s0001                                       scaffold_994   \n",
       "Mapoly0998s0001  scaffold_998;scaffold_998;scaffold_998;scaffol...   \n",
       "\n",
       "                                                        Start  \\\n",
       "Geneid                                                          \n",
       "MapolyY_A0001                                           10168   \n",
       "MapolyY_A0002                                           14752   \n",
       "MapolyY_A0003                                           69760   \n",
       "MapolyY_A0004             74668;75039;75534;75986;84872;84872   \n",
       "MapolyY_A0005                                          125980   \n",
       "...                                                       ...   \n",
       "Mapoly0099s0063                   770737;771728;772563;773007   \n",
       "Mapoly0991s0001                                          5953   \n",
       "Mapoly0992s0001                          11;547;996;1312;1839   \n",
       "Mapoly0994s0001                                          4875   \n",
       "Mapoly0998s0001  1631;1804;2065;2469;2959;3516;3881;4406;4625   \n",
       "\n",
       "                                                          End  \\\n",
       "Geneid                                                          \n",
       "MapolyY_A0001                                           10935   \n",
       "MapolyY_A0002                                           16054   \n",
       "MapolyY_A0003                                           71066   \n",
       "MapolyY_A0004             74870;75338;75600;77256;86506;86506   \n",
       "MapolyY_A0005                                          126717   \n",
       "...                                                       ...   \n",
       "Mapoly0099s0063                   771554;772149;772759;775346   \n",
       "Mapoly0991s0001                                          6171   \n",
       "Mapoly0992s0001                        255;712;1187;1652;1883   \n",
       "Mapoly0994s0001                                          6206   \n",
       "Mapoly0998s0001  1690;1859;2102;2805;3334;3733;4085;4588;4861   \n",
       "\n",
       "                            Strand  Length  \n",
       "Geneid                                      \n",
       "MapolyY_A0001                    -     768  \n",
       "MapolyY_A0002                    -    1303  \n",
       "MapolyY_A0003                    -    1307  \n",
       "MapolyY_A0004          +;+;+;+;+;+    3476  \n",
       "MapolyY_A0005                    -     738  \n",
       "...                            ...     ...  \n",
       "Mapoly0099s0063            -;-;-;-    3777  \n",
       "Mapoly0991s0001                  -     219  \n",
       "Mapoly0992s0001          -;-;-;-;-     989  \n",
       "Mapoly0994s0001                  -    1332  \n",
       "Mapoly0998s0001  +;+;+;+;+;+;+;+;+    1710  \n",
       "\n",
       "[19287 rows x 5 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.var"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## gene annotation\n",
    "\n",
    "- index = ID\n",
    "- name = str\n",
    "- TF, ethylen, ABA, JA, cytokinin, auxin, GA, strigolactone, MIR = TRUE or FALSE\n",
    "- id = some GO or PFRAM etc., ID\n",
    "- idtype = id type\n",
    "- description = description\n",
    "- Lenght = exon Lenght"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_annotations = pd.read_csv(\"../data/annotation/gene_annotations.csv\", index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>TF</th>\n",
       "      <th>ethylene</th>\n",
       "      <th>ABA</th>\n",
       "      <th>JA</th>\n",
       "      <th>cytokinin</th>\n",
       "      <th>auxin</th>\n",
       "      <th>GA</th>\n",
       "      <th>strigolactone</th>\n",
       "      <th>MIR</th>\n",
       "      <th>Id</th>\n",
       "      <th>IdType</th>\n",
       "      <th>description</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Mapoly0001s0001</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>GO:0006397 PF01348</td>\n",
       "      <td>GO PFAM</td>\n",
       "      <td>mRNA processing Type II intron maturase</td>\n",
       "      <td>249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0001s0002</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>PF00628 GO:0005515</td>\n",
       "      <td>PFAM GO</td>\n",
       "      <td>PHD-finger protein binding</td>\n",
       "      <td>660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0001s0003</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0001s0004</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>PTHR12459:SF1 PTHR12459</td>\n",
       "      <td>PANTHER PANTHER</td>\n",
       "      <td>gb def: k02g10.3.p [caenorhabditis elegans] UN...</td>\n",
       "      <td>1248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mapoly0001s0005</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>PTHR23272</td>\n",
       "      <td>PANTHER</td>\n",
       "      <td>BED FINGER-RELATED</td>\n",
       "      <td>792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_B0046</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_B0047</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_B0048</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>PTHR21004 PTHR21004:SF0</td>\n",
       "      <td>PANTHER PANTHER</td>\n",
       "      <td>SERINE PROTEASE-RELATED SUBFAMILY NOT NAMED</td>\n",
       "      <td>1810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_B0049</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MapolyY_B0050</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>GO:0008270 PF00098 GO:0003676</td>\n",
       "      <td>GO PFAM GO</td>\n",
       "      <td>zinc ion binding Zinc knuckle nucleic acid bin...</td>\n",
       "      <td>1591</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19287 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  name     TF  ethylene    ABA     JA  cytokinin  auxin  \\\n",
       "Mapoly0001s0001  False  False     False  False  False      False  False   \n",
       "Mapoly0001s0002  False  False     False  False  False      False  False   \n",
       "Mapoly0001s0003  False  False     False  False  False      False  False   \n",
       "Mapoly0001s0004  False  False     False  False  False      False  False   \n",
       "Mapoly0001s0005  False  False     False  False  False      False  False   \n",
       "...                ...    ...       ...    ...    ...        ...    ...   \n",
       "MapolyY_B0046    False  False     False  False  False      False  False   \n",
       "MapolyY_B0047    False  False     False  False  False      False  False   \n",
       "MapolyY_B0048    False  False     False  False  False      False  False   \n",
       "MapolyY_B0049    False  False     False  False  False      False  False   \n",
       "MapolyY_B0050    False  False     False  False  False      False  False   \n",
       "\n",
       "                    GA  strigolactone    MIR                             Id  \\\n",
       "Mapoly0001s0001  False          False  False             GO:0006397 PF01348   \n",
       "Mapoly0001s0002  False          False  False             PF00628 GO:0005515   \n",
       "Mapoly0001s0003  False          False  False                           None   \n",
       "Mapoly0001s0004  False          False  False        PTHR12459:SF1 PTHR12459   \n",
       "Mapoly0001s0005  False          False  False                      PTHR23272   \n",
       "...                ...            ...    ...                            ...   \n",
       "MapolyY_B0046    False          False  False                           None   \n",
       "MapolyY_B0047    False          False  False                           None   \n",
       "MapolyY_B0048    False          False  False        PTHR21004 PTHR21004:SF0   \n",
       "MapolyY_B0049    False          False  False                           None   \n",
       "MapolyY_B0050    False          False  False  GO:0008270 PF00098 GO:0003676   \n",
       "\n",
       "                          IdType  \\\n",
       "Mapoly0001s0001          GO PFAM   \n",
       "Mapoly0001s0002          PFAM GO   \n",
       "Mapoly0001s0003             None   \n",
       "Mapoly0001s0004  PANTHER PANTHER   \n",
       "Mapoly0001s0005          PANTHER   \n",
       "...                          ...   \n",
       "MapolyY_B0046               None   \n",
       "MapolyY_B0047               None   \n",
       "MapolyY_B0048    PANTHER PANTHER   \n",
       "MapolyY_B0049               None   \n",
       "MapolyY_B0050         GO PFAM GO   \n",
       "\n",
       "                                                       description  Length  \n",
       "Mapoly0001s0001            mRNA processing Type II intron maturase     249  \n",
       "Mapoly0001s0002                         PHD-finger protein binding     660  \n",
       "Mapoly0001s0003                                               None    1770  \n",
       "Mapoly0001s0004  gb def: k02g10.3.p [caenorhabditis elegans] UN...    1248  \n",
       "Mapoly0001s0005                                 BED FINGER-RELATED     792  \n",
       "...                                                            ...     ...  \n",
       "MapolyY_B0046                                                 None     384  \n",
       "MapolyY_B0047                                                 None    1604  \n",
       "MapolyY_B0048          SERINE PROTEASE-RELATED SUBFAMILY NOT NAMED    1810  \n",
       "MapolyY_B0049                                                 None    3955  \n",
       "MapolyY_B0050    zinc ion binding Zinc knuckle nucleic acid bin...    1591  \n",
       "\n",
       "[19287 rows x 14 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "import anndata\n",
      "import pandas as pd\n",
      "from reader import read_featureCounts as reader\n",
      "import sys\n",
      "\n",
      "sys.path.append(\"../ngsutils/\")\n",
      "\n",
      "print(sys.path)\n",
      "import sys\n",
      "sys.path.append(\"../ngsutils/\")\n",
      "print(sys.path)\n",
      "\n",
      "import anndata\n",
      "import pandas as pd\n",
      "from reader import read_featureCounts as reader\n",
      "import sys\n",
      "sys.path.append(\"../ngsutils/\")\n",
      "print(sys.path)\n",
      "\n",
      "import anndata\n",
      "import pandas as pd\n",
      "import reader\n",
      "import sys\n",
      "sys.path.append(\"../ngsutils/\")\n",
      "sys.path.append(\"../ngsutils/reader\")\n",
      "print(sys.path)\n",
      "\n",
      "import anndata\n",
      "import pandas as pd\n",
      "import reader\n",
      "path_to_featureCounts_data = \"../data/read_featureCounts/featureCounts_sample.txt\"\n",
      "path_to_conditions = \"../data/read_featureCounts/conditions.txt\"\n",
      "adata = reader.read_featureCounts(path_to_featureCounts_data, path_to_conditions)\n",
      "adata\n",
      "adata.var\n",
      "gene_id = pd.read_csv(\"../data/annotation/geneID.csv\", index_col=0)\n",
      "function = pd.read_csv(\"../data/annotation/gene_function.csv\", index_col=0)\n",
      "regulatory_genes = pd.read_csv(\"../data/annotation/regulatory_genes.csv\", index_col=0)\n",
      "regulatory_genes\n",
      "regulatory_genes['name', 'category']\n",
      "regulatory_genes\n",
      "regulatory_genes['name', 'category']\n",
      "regulatory_genes[['name', 'category']]\n",
      "regulatory_genes = regulatory_genes[['name', 'category']]\n",
      "gene_annotation = pd.concat((gene_id, function), axis=1).fillna(\"None\")\n",
      "pd.concat((gene_annotation, regulatory_genes), axis=1)\n",
      "regulatory_genes[regulatory_genes.duplicated()]\n",
      "regulatory_genes.drop_duplicates()\n",
      "regulatory_genes\n",
      "duplicate = regulatory_genes[regulatory_genes.duplicated()]\n",
      "regulatory_genes.drop_duplicates()\n",
      "regulatory_genes = regulatory_genes.drop_duplicates()\n",
      "regulatory_genes.duplicated()\n",
      "regulatory_genes.duplicated().sum()\n",
      "pd.concat((regulatory_genes, gene_annotation))\n",
      "pd.concat((regulatory_genes, gene_annotation), axis=1)\n",
      "regulatory_genes[regulatory_genes['category'] == 'TF']\n",
      "TF = regulatory_genes[regulatory_genes['category'] == 'TF']\n",
      "pd.concat((TF, gene_annotation), axis=1)\n",
      "set(regulatory_genes['category'])\n",
      "TF = regulatory_genes['category'] == 'TF'\n",
      "TF\n",
      "for category in set(regulatory_genes['category']):\n",
      "    print(category)\n",
      "category_df = pd.DataFrame()\n",
      "for category in set(regulatory_genes['category']):\n",
      "    tmp = regulatory_genes['category'] == category\n",
      "    category_df = pd.concat(category_df, tmp)\n",
      "category_df = pd.DataFrame()\n",
      "for category in set(regulatory_genes['category']):\n",
      "    tmp = regulatory_genes['category'] == category\n",
      "    category_df = pd.concat((category_df, tmp), axis=1)\n",
      "category_df = pd.DataFrame()\n",
      "for category in set(regulatory_genes['category']):\\\n",
      "    print(category)\n",
      "    tmp = regulatory_genes['category'] == category\n",
      "    category_df = pd.concat((category_df, tmp), axis=1)\n",
      "category_df = pd.DataFrame()\n",
      "for category in set(regulatory_genes['category']):\n",
      "    print(category)\n",
      "    tmp = regulatory_genes['category'] == category\n",
      "    category_df = pd.concat((category_df, tmp), axis=1)\n",
      "category\n",
      "category_df\n",
      "category_dict = {}\n",
      "for category in set(regulatory_genes['category']):\n",
      "    tmp = regulatory_genes['category'] == category\n",
      "    category_dict['category'] = tmp\n",
      "category_dict\n",
      "category_dict = {}\n",
      "for category in set(regulatory_genes['category']):\n",
      "    tmp = regulatory_genes['category'] == category\n",
      "    category_dict[category] = tmp\n",
      "category_dict['']\n",
      "category_dict['TF']\n",
      "for k, v in category_dict.items():\n",
      "    print(k)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylent']), axis=1)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylen']), axis=1)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylene']), axis=1)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylene'], category_dict['auxin']), axis=1)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylene'], category_dict['auxin'], category_dict['JA']), axis=1)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylene'], category_dict['auxin'], category_dict['JA'], category_dict['ABA']), axis=1)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylene'], category_dict['auxin'], category_dict['JA'], category_dict['ABA'], category_dict['cytokinin']), axis=1)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylene'], category_dict['auxin'], category_dict['JA'], category_dict['ABA'], category_dict['cytokinin'], category_dict['GA']), axis=1)\n",
      "pd.concat((category_dict['TF'], category_dict['ethylene'], category_dict['auxin'], category_dict['JA'], category_dict['ABA'], category_dict['cytokinin'], category_dict['GA'], category_dict['strigolactone']), axis=1)\n",
      "tmp = category_dict['TF']\n",
      "tmp.rename(columns=\"TF\")\n",
      "category_df = pd.DataFrame()\n",
      "for k, v in category_dict.items():\n",
      "    if k == 'MIR':\n",
      "        continue\n",
      "    category_df = pd.concat((category_df, v), aixs=1)\n",
      "category_df = pd.DataFrame()\n",
      "for k, v in category_dict.items():\n",
      "    if k == 'MIR':\n",
      "        continue\n",
      "    category_df = pd.concat((category_df, v), axis=1)\n",
      "category_dict\n",
      "pd.DataFrame.from_dict(category_dict)\n",
      "regulatory_df = pd.DataFrame.from_dict(category_dict)\n",
      "regulatory_df = pd.concat((regulatory_df, regulatory_genes['name']), axis=1)\n",
      "regulatory_df\n",
      "regulatory_df.columns\n",
      "regulatory_df = regulatory_df[['name', 'TF', 'ethylene', 'ABA', 'JA', 'cytokinin', 'auxin', 'GA', 'strigolactone', 'MIR']]\n",
      "pd.concat(gene_annotation, regulatory_df)\n",
      "pd.concat((gene_annotation, regulatory_df), axis=1)\n",
      "regulatory_df.loc[~regulatory_df.index.str.contains(\"CUFF\")]\n",
      "regulatory_df.loc[~regulatory_df.index.str.contains(\"CUFF\")].drop_duplicates()\n",
      "regulatory_df\n",
      "regulatory_df = regulatory_df.loc[~regulatory_df.index.str.contains(\"CUFF\")].drop_duplicates()\n",
      "pd.concate(regulatory_df, gene_annotatin)\n",
      "pd.concat((regulatory_df, gene_annotatin), axis=1)\n",
      "pd.concat((regulatory_df, gene_annotatins), axis=1)\n",
      "pd.concat((regulatory_df, gene_annotation), axis=1)\n",
      "gene_annotation\n",
      "gene_id\n",
      "gene_id = pd.read_csv(\"../data/annotation/geneID.csv\", index_col=0, header=None)\n",
      "function = pd.read_csv(\"../data/annotation/gene_function.csv\", index_col=0)\n",
      "regulatory_genes = pd.read_csv(\"../data/annotation/regulatory_genes.csv\", index_col=0)\n",
      "gene_id\n",
      "pd.concat((gene_id, regulatory_df), axis=1)\n",
      "for idx in regulatory_df.index:\n",
      "    if idx not in gene_id.index:\n",
      "        print(idx)\n",
      "len(set(regulatory_df.index))\n",
      "regulatory_df[regulatory_df.duplicated()]\n",
      "regulatory_df[regulatory_df.duplicated(subset='ID')]\n",
      "idxs = []\n",
      "for idx in regulatory_df.index:\n",
      "    if idx not in idxs:\n",
      "        idxs.append(idx)\n",
      "    else:\n",
      "        print(regulatory_df.loc[idx])\n",
      "regulatory_df.to_csv(\"regulatory_genes.csv\")\n",
      "regulatory_df = pd.read_csv(\"regulatory_genes.csv\")\n",
      "pd.concate((gene_id, regulatory_df))\n",
      "pd.concat((gene_id, regulatory_df), axis=1)\n",
      "regulatory_df = pd.read_csv(\"regulatory_genes.csv\", index_col=0)\n",
      "pd.concat((gene_id, regulatory_df), axis=1)\n",
      "pd.concat((gene_id, regulatory_df), axis=1, sort=True)\n",
      "regulatory_df\n",
      "pd.concat((gene_id, regulatory_df), axis=1, sort=True).fillna(False)\n",
      "df = pd.concat((gene_id, regulatory_df), axis=1, sort=True).fillna(False)\n",
      "pd.concat((df, function), axis=1, sort=True)\n",
      "pd.concat((df, function), axis=1, sort=True).fillna(None)\n",
      "pd.concat((df, function), axis=1, sort=True).fillna(\"None\")\n",
      "df = pd.concat((df, function), axis=1, sort=True).fillna(\"None\")\n",
      "df.replace({'name': {False: 'No Name'}})\n",
      "df.replace({'name': {False: 'No Name'}}).to_csv(\"gene_annotations.csv\")\n",
      "gene_id = pd.read_csv(\"../data/annotation/geneID.csv\", index_col=0, header=None)\n",
      "function = pd.read_csv(\"../data/annotation/gene_function.csv\", index_col=0)\n",
      "regulatory_genes = pd.read_csv(\"../data/annotation/regulatory_genes.csv\", index_col=0)\n",
      "gene_length = pd.read_csv('../data/annotation/geneLength.csv', index_col=0)\n",
      "pd.concat((df, gene_length), axis=1, sort=True)\n",
      "gene_length\n",
      "gene_length.set_index(gene_id)\n",
      "gene_length.set_index(list(gene_id))\n",
      "adata.obs\n",
      "adata.var\n",
      "pd.concat((df, adata.var['Length']), axis=1, sort=True)\n",
      "pd.concat((df, adata.var['Length']), axis=1, sort=True).to_csv(\"gene_annotations.csv\")\n",
      "gene_annotations = pd.read_csv(\"../data/annotation/gene_annotation.csv\")\n",
      "gene_annotations = pd.read_csv(\"../data/annotation/gene_annotations.csv\", index_col=0)\n",
      "gene_annotation\n",
      "gene_annotations\n",
      "gene_annotations.replace({\"name\": {False: \"No Name\"}})\n",
      "gene_annotations.replace({\"name\": {False: \"No Name\"}}).to_csv(\"../data/annotation/gene_annotations.csv\")\n",
      "gene_annotations.replace({\"name\": {False: \"No Name\"}}).to_csv(\"../data/annotation/gene_annotations.csv\")\n",
      "gene_annotations = pd.read_csv(\"../data/annotation/gene_annotations.csv\", index_col=0)\n",
      "gene_annotations\n",
      "%history\n"
     ]
    }
   ],
   "source": [
    "%history"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_annotations.replace({\"name\": {False: \"No Name\"}}).to_csv(\"../data/annotation/gene_annotations.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
